hal/include/HAL/Encoder.h

```diff
@@ -9,26 +9,57 @@
 
 #include <stdint.h>
 
+#include "HAL/Handles.h"
+
 extern "C" {
-void* initializeEncoder(uint8_t port_a_module, uint32_t port_a_pin,
-                        bool port_a_analog_trigger, uint8_t port_b_module,
-                        uint32_t port_b_pin, bool port_b_analog_trigger,
-                        bool reverseDirection, int32_t* index,
-                        int32_t* status);  // TODO: fix routing
-void freeEncoder(void* encoder_pointer, int32_t* status);
-void resetEncoder(void* encoder_pointer, int32_t* status);
-int32_t getEncoder(void* encoder_pointer, int32_t* status);  // Raw value
-double getEncoderPeriod(void* encoder_pointer, int32_t* status);
-void setEncoderMaxPeriod(void* encoder_pointer, double maxPeriod,
-                         int32_t* status);
-bool getEncoderStopped(void* encoder_pointer, int32_t* status);
-bool getEncoderDirection(void* encoder_pointer, int32_t* status);
-void setEncoderReverseDirection(void* encoder_pointer, bool reverseDirection,
+enum EncoderIndexingType {
+  kResetWhileHigh,
+  kResetWhileLow,
+  kResetOnFallingEdge,
+  kResetOnRisingEdge
+};
+enum EncoderEncodingType { k1X, k2X, k4X };
```

Comment:
Well they need to be accessible from the C API. There are other places (namely counter and the NetComm code) that use global enum names. They'll all probably be prefixed HAL when we go do that prefix rewrite.